launcher/ui/dialogs/ExportPackDialog.cpp

```diff
@@ -29,104 +29,110 @@
 
-        ui->summary->setPlainText(instance->settings()->get("ExportSummary").toString());
+        m_ui->summary->setPlainText(instance->settings()->get("ExportSummary").toString());
     } else {
         setWindowTitle(tr("Export CurseForge Pack"));
 
-        ui->summaryLabel->hide();
-        ui->summary->hide();
+        m_ui->summaryLabel->hide();
+        m_ui->summary->hide();
 
-        ui->author->setText(instance->settings()->get("ExportAuthor").toString());
+        m_ui->author->setText(instance->settings()->get("ExportAuthor").toString());
     }
 
     // ensure a valid pack is generated
     // the name and version fields mustn't be empty
-    connect(ui->name, &QLineEdit::textEdited, this, &ExportPackDialog::validate);
-    connect(ui->version, &QLineEdit::textEdited, this, &ExportPackDialog::validate);
+    connect(m_ui->name, &QLineEdit::textEdited, this, &ExportPackDialog::validate);
+    connect(m_ui->version, &QLineEdit::textEdited, this, &ExportPackDialog::validate);
     // the instance name can technically be empty
     validate();
 
     QFileSystemModel* model = new QFileSystemModel(this);
-    model->setIconProvider(&icons);
+    model->setIconProvider(&m_icons);
 
     // use the game root - everything outside cannot be exported
-    const QDir root(instance->gameRoot());
-    proxy = new FileIgnoreProxy(instance->gameRoot(), this);
-    proxy->ignoreFilesWithPath().insert({ "logs", "crash-reports", ".cache", ".fabric", ".quilt" });
-    proxy->ignoreFilesWithName().append({ ".DS_Store", "thumbs.db", "Thumbs.db" });
-    proxy->setSourceModel(model);
+    const QDir instanceRoot(instance->instanceRoot());
+    m_proxy = new FileIgnoreProxy(instance->instanceRoot(), this);
+    auto prefix = QDir(instance->instanceRoot()).relativeFilePath(instance->gameRoot());
+    for (auto path : { "logs", "crash-reports", ".cache", ".fabric", ".quilt" }) {
+        m_proxy->ignoreFilesWithPath().insert(FS::PathCombine(prefix, path));
+    }
+    m_proxy->ignoreFilesWithName().append({ ".DS_Store", "thumbs.db", "Thumbs.db" });
+    m_proxy->setSourceModel(model);
+    loadPackIgnore();
 
     const QDir::Filters filter(QDir::AllEntries | QDir::NoDotAndDotDot | QDir::AllDirs | QDir::Hidden);
-
-    for (const QString& file : root.entryList(filter)) {
-        if (!(file == "mods" || file == "coremods" || file == "datapacks" || file == "config" || file == "options.txt" ||
-              file == "servers.dat"))
-            proxy->blockedPaths().insert(file);
+    if (m_proxy->blockedPaths().leaf()) {  // only add this if the list is empty
+        const QDir gameRoot(instance->gameRoot());
+        for (const QString& file : gameRoot.entryList(filter)) {
+            if (!(file == "mods" || file == "coremods" || file == "datapacks" || file == "config" || file == "options.txt" ||
+                  file == "servers.dat"))
+                m_proxy->blockedPaths().insert(FS::PathCombine(prefix, file));
+        }
```

Comment:
If we're using one file for both instance export and modpack export, it would be best to just get rid of this. It's pretty broken anyway, new files/dirs would be checked by default if they were added after opening the dialog the first time, and opening the dialog before the instance zip export would affect it

Reply:
The reason I closed this PR is the implementation would need to be completely different to keep this functionality otherwise it's quite buggy

Reply:
Removed the if for now.
There is no way to make both the default checked folders/files and packignore work together.
You are right if we want to implement this as you want we need to keep a list with the default checked files.
But that way it will either behave differently compared with the prism export or break the packignore file that we have right now.